bot/resources/tags/venv.md

```diff
@@ -0,0 +1,21 @@
+**Virtual Environments**
+
+Virtual environments are isolated Python environments, which make it easier to keep your system clean and manage dependencies. By default, when activated, only libraries and scripts installed in the virtual environment are accessible.
+
+To create a new virtual environment, you can use the standard library `venv` module: `python3 -m venv .venv` (replace `python3` with `python` or `py` on Windows)
+
+Then, to activate the new virtual environment:
+
+PowerShell: `.venv\Scripts\Activate.ps1`
+Command Prompt: `.venv\Scripts\activate.bat`
+Bash: `source .venv/bin/activate`
```

Comment:
We should note that bash applies to linux/osx. It took me a moment ( despite extensive familiarity with all three here and the os's and python ) to realize that yeah, the line marked "bash" was the one that I was looking for. If we put (OSX/Linux) in there, or perhaps something better if you can figure it out, that might help make things clearer and more obvious.

Reply:
My concern with specifying an OS would be that bash, for example, could be for Linux, OSX or WSL. Likewise, PowerShell can be installed on platforms other than Windows. The only specific difference between the three is the shell itself.

I guess this is a little pedantic, and can probably be ignored. Thoughts?

Reply:
WSL is linux, technically. While you can get bash underneath windows proper with MingW, the reason to mention the OS is to catch the people who are using it (starting anew on mac, or just getting their feet wet on linux), who think of "terminal" rather than "bash". 

It is a significant portion of the people likely to view the tag, rather than those who've installed powershell on something other than windows, who are unlikely to get confused on which applies since they've chosen to install the tech by name manually. 

If we did have a linux distro of some popularity utilizing powershell as the default we'd want to note that here. There isn't one, so thankfully we won't be tempted to.

Reply:
> If we did have a linux distro of some popularity utilizing powershell as the default

Sent shudders down my spine haha

Yeah, with the target audience, it's probably best to choose clarity over accuracy. Would something like below be okay?
> Shell (OS): `path/to/activate`

Reply:
Well, I'm not sure what the best option is, but again, we're painting targets. Like how bow targets are concentric rings and the comment button here is green and on the bottom right, we want the people searching the tag for what activate command they need to have their eyes immediately go to the right spot. 

I do think duplicating Windows (powershell) might be too much.. but maybe not. I'll defer to other's judgement on that.

Reply:
Writing "Windows" twice I think would be redundant. I've used bold for the OS to draw attention to the determining factor. This is the best way I can think to distinguish between PowerShell and Command Prompt:

**Windows** (PowerShell): `.venv\Scripts\Activate.ps1`
or (Command Prompt): `.venv\Scripts\activate.bat`
**MacOS / Linux** (Bash): `source .venv/bin/activate`

Reply:
I think that sounds pretty good and clear.